src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -90,9 +91,16 @@ public SliceFromConfig(final RepoConfig config, final Vertx vertx,
                         "file", config -> CompletableFuture.completedStage(new FilesSlice(storage, permissions, auth))
                     ),
                     new MapEntry<>(
-                        "npm", config -> CompletableFuture.completedStage(
-                        new NpmSlice(new Npm(storage), storage)
-                    )
+                        "npm", config -> config.path().thenCombine(
+                            config.settings(),
+                            (path, settings) -> new NpmSlice(
+                                path,
+                                new Npm(storage),
```

Comment:
@brastak why do we need both: `storage` and `Npm(storage)`? It seems we can pass only storage and construct `Npm(storage)` internally in `NpmSlice`, WDYT?

Reply:
@g4s8 It's previous design that I have not changed. Currently, `UploadSlice` saves a file directly to `Storage`, then `Npm`  processes the uploaded file and saves transformed metadata to `Storage`, then `UploadSlice` removes an original uploaded file from `Storage`.

Really, such design makes sense because of `Npm` is used by our clients and it's their responsibility to put initial file to the right place. But I can add `todo` and introduce all processing cycle into `Npm`